### DISCUSSION TITLE
Régebbi adatok visszadolgozása

### DISCUSSION THREAD

**Discussion by ZsBT** [ZsBT/hass-w1000-portal#15] [General]
Összefoglalva, a fontos pontok:

**1) legyen megadható, hány napra visszamenőleg legyenek beolvasva az adatok**

Ez jeleneg kettő nap ( = tegnap és tegnapelőtt). Általában elegendő, mert az EON adatok nem szoktak késni ennél többet. Előfordul azonban, hogy utólag derül ki hogy bizonyos napok adatai nem voltak pontosak, esetleg "ősfeltöltést" kéne végezni. 
Ilyenkor jó lenne, ha pl. HA konfig yaml paraméterben megadhatnánk hogy hány napról legyen szó. Ehhez értelemszerűen hozzá kell igazítani az EON portál riportját.
Fontos tudni, hogy ezt csak ideiglenesen használhatjuk, mert a szolgáltató adatfeldolgozó részlege érzékeny a túl sok, vagy túl sűrű adatletöltésre (ebbe javaslom törődjünk bele, nem a mi dolgunk felülbírálni hogy mit bír a w1000 portál, vagy hogy lehetne javítani rajta). A lényeg, hogy napi 4-5 lekérdezésre kell szorítkoznunk.
Ez a feature könnyen kivitelezhető, opcionális paraméterként bekerülhet a HA konfigba.

**2) az utolsó napon óriási tüske jelentkezik a fogyasztásban és termelésben**

Ez pedig amiatt van, mert a HA szenzorhoz tartozó visszamenőleges statisztikai adatokat felülcsapjuk, azonban e szenzor értékét (ami maga az óraállás) a HA aktuális fogyasztásnak rögzíti. Tehát az utolsó nap utolsó órájára nem pl 3kWh-t, hanem 15200kWh-t statisztikáz be. Ez az érték persze a következő olvasásnál felülírásra kerül, de akkor is szétbassza az összes grafikont.

Ehhez masszívan át kell írni az integrációt és hozzá át is kell alakítani mindenkinek a saját riportjait az EON portálon. Ez többé-kevésbé már megvan nálam, de úgy látom többen is vannak akik szívesen tesztelnék. Ennek  nagyon örülök!
A terv, hogy a DEV ágba behúzom az új verziót, ahonnan mindenki letöltheti és felülírhatja a komponensét. Az új szenzor azonban már nem fogja tartalmazni az óraállást, a szenzor értéke az utolsó óra fogyasztási értéke lesz, az óraállás meg vagy attribútumban, vagy egy tök más szenzorban jelenhet meg.

Ezeket látom most, ha kihagytam valamit, jelezzétek.
Itt fogok szólni, ha a DEV ágban elérhető az új verzió.

Zsombor

**Comment by toranis** [comment]
Szia Zsonbor! Tetszik az irány, de: mi lenne ha a telepítés előtti napra tudnál korábbi adatot behúzni egy eon nem HA-ban végzett, hanem távleolvasás weboldalról letöltött "offline" csv vagy xls exporottal? Nem lenne egyszerűbb? Akkor az óraállás érték gyakorlatilag irreleváns lenne, elég lenne egy "zéró" állapotot beállítani" ami pedig megadható érték lenne. (sajna csak a logikát igyekszem leírni, mert programozni nem tudok)

**Comment by ZsBT** [comment]
Igen, a terv az, hogy az óraállás már nem lesz a új verzióban. Ez az
integráció az EON portál letöltésére hivatott, egy csv importhoz ez nem
szükséges - ilyen funkcióval ebben a projektben egyelőre nem foglalkozom.

**Comment by NorbertT83** [comment]
Sziasztok! Lehet én néztem be valamit és nem találtam meg a módját, de nálam az EOn egész napos adat nyilván az azt követő napon jelenik meg, mert addigra lesz végleges. Viszont így teljesen hazavágja az aznapi napenergia termeléshez való viszonyát, mivel az pedig azonnali adat. (A tegnap exportját veti össze az Energy panel a ma termelésével) Nem lehetne esetleg a tegnap adatait (amit ma kapok meg) eleve "visszadátumoztatni" tegnapra?

**Comment by ZsBT** [comment]
https://github.com/Netesfiu/EON-SPOOKER
